Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
Pushed :+1:.

Benchmarks:
* brew  0.18s user 0.05s system 97% cpu 0.233 total
* brew  0.18s user 0.05s system 97% cpu 0.229 total
* brew  0.17s user 0.05s system 98% cpu 0.226 total
* brew  0.18s user 0.05s system 98% cpu 0.233 total
* brew  0.18s user 0.05s system 98% cpu 0.230 total

Sharing the check knocked a tiny bit of the average there, which is a nice upside.